src/public/providers.ts

```diff
@@ -32,14 +32,17 @@ class DomainElement {
         <div class="col-11">
           <span class="domainElement">${domainObj.domain}</span>
         </div>
-        <div class="col-1">
-        <button type="button" class="btn btn-primary setUpButton">Setup</button>
+        <div class="col-1 actionContainer">
+        <button type="button" class="btn btn-primary setUpButton" type="button">Setup</button>
+        <button class='btn btn-primaryloading loading'>Loading...</button>
         </div>
       </div>
     `
     domainElement.classList.add('list-group-item')
+    const actionContainer: HTMLElement = helper.getElement('.actionContainer', domainElement)
     const setUpButton = helper.getElement('.setUpButton', domainElement)
     setUpButton.onclick = (): void => {
+      actionContainer.classList.add('addLoading')
```

Comment:
`addLoading` does not sound like a good className, because it sounds more like a function. `loading` or `isLoading` would make more sense.